generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs

```diff
@@ -0,0 +1,93 @@
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+import { Component, ViewChild, OnInit, OnDestroy, AfterViewInit, ElementRef } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { LoginService } from 'app/core/login/login.service';
+import { AccountService } from 'app/core/auth/account.service';
+import { Account } from 'app/core/user/account.model';
+
+@Component({
+  selector: '<%= jhiPrefixDashed %>-login',
+  templateUrl: './login.component.html'
+})
+export class LoginComponent implements OnInit, AfterViewInit, OnDestroy {
+  account: Account | null = null;
+  @ViewChild('username', { static: false })
+  username?: ElementRef;
+
+  authSubscription?: Subscription;
+  authenticationError = false;
+
+  loginForm = this.fb.group({
+    username: [null, [Validators.required]],
+    password: [null, [Validators.required]],
+    rememberMe: [false]
+  });
+
+  constructor(private accountService: AccountService, private loginService: LoginService, private router: Router, private fb: FormBuilder) {}
+
+  ngOnInit(): void {
+    // try to log in automatically
+    this.authSubscription = this.accountService.identity(true).subscribe(account => {
+      this.account = account;
+      if (this.isAuthenticated()) {
+        this.router.navigate(['']);
+      }
+    });
+  }
+
+  ngAfterViewInit(): void {
+    if (this.username) {
+      this.username.nativeElement.focus();
+    }
+  }
+
+  login(): void {
+    this.loginService
+      .login({
+        username: this.loginForm.get('username')!.value,
+        password: this.loginForm.get('password')!.value,
+        rememberMe: this.loginForm.get('rememberMe')!.value,
+      })
+      .subscribe(
+        () => {
+          this.authenticationError = false;          
+          if (!this.router.getCurrentNavigation()) {
+            // There were no routing during login (eg from navigationToStoredUrl)
+            this.router.navigate(['']);
+         }
+        },
+        () => {
+          this.authenticationError = true;
+        }
+      );
+  }
+
+  isAuthenticated(): boolean {
+    return this.accountService.isAuthenticated();
+  }
+
+  ngOnDestroy(): void {
+    if (this.authSubscription) {
+      this.authSubscription.unsubscribe();
+    }
+  }
```

Comment:
One more note. As `accountService.identity()` returns `Observable` from `HttpClient` then there is no need to `unsubscribe` because `HttpClient` completes automatically after emitting value. `unsubscribe` doesn't harm but currently we are not unsubscribing from `HttpClient` observables. To keep this style can you remove unsubscription and related variables?
```suggestion
```